### PR TITLE
Increase max_rows_to_read limit in some tests

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Increase max_rows_to_read limit in some tests.

Some tests reading from system.text_log, such as `02796_calculate_text_stack_trace`, `02581_share_big_sets_between_mutation_tasks_long`, `02581_share_big_sets_between_multiple_mutations_tasks_long` fail spuriously as they reach this limit.

Introduced in https://github.com/ClickHouse/ClickHouse/pull/66837

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 9edbffef841a3e34e17d4e0c193c050d9826ba42 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70617/9edbffef841a3e34e17d4e0c193c050d9826ba42/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/df4e80bd01c4baaba95cd9471a7f6b95ac5db465/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/integration_tests__asan__old_analyzer__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/733e61cfab8e7a99d0ea5b685218a9db98de3568/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by nickitat — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Can we change it inside tests that read from text_log?

**Comment by nickitat** [comment]
IMO it either should be global or we need a check like the one for presence of filtering by `currentDatabase()`, otherwise people will continue to reproduce this issue in every second test.

File: tests/clickhouse-test [hunk 1]
```diff
@@ -292,6 +292,7 @@ def clickhouse_execute_http(
         "http_receive_timeout": timeout,
         "http_send_timeout": timeout,
         "output_format_parallel_formatting": 0,
+        "max_rows_to_read": 0,  # Some queries read from system.text_log which might get too big
```
Comment (alexey-milovidov) [inline tests/clickhouse-test:295]: Ok. Although I don't understand this change - if we unlock it in some tests, what is the point for unlocking it globally here?

PS. Anyways, the changes LGTM. Thank you!

**Review by alexey-milovidov — APPROVED** [review]

File: tests/clickhouse-test (hunk 1 above)
Comment (Algunenano) [reply tests/clickhouse-test:295]: AFAIK this is not global, it's only used when executing SQL queries inside clickhouse-test (for internal usage).

**Pull request merged**